src/components/ai-input.tsx

```diff
@@ -29,12 +29,48 @@ export const AIInput = ( {
 }: AIInputProps ) => {
 	const inputRef = useRef< HTMLTextAreaElement >( null );
 
+	const [ isTyping, setIsTyping ] = useState( false );
+	const typingTimeout = useRef< NodeJS.Timeout >();
+
+	const { RiveComponent, inactiveInput, thinkingInput, typingInput, startStateMachine } =
+		useAiIcon();
+
 	useEffect( () => {
 		if ( ! disabled && inputRef.current ) {
 			inputRef.current.focus();
 		}
 	}, [ disabled ] );
 
+	useEffect( () => {
+		startStateMachine();
+
+		if ( inactiveInput ) {
+			inactiveInput.value = disabled;
+		}
+
+		if ( thinkingInput ) {
+			thinkingInput.value = isAssistantThinking;
+		}
+
+		if ( typingInput ) {
+			typingInput.value = isTyping;
+		}
+	}, [
+		isAssistantThinking,
+		isTyping,
+		disabled,
+		startStateMachine,
+		inactiveInput,
+		thinkingInput,
+		typingInput,
+	] );
+
+	useEffect( () => () => {
+		if ( typingTimeout.current ) {
+			clearTimeout( typingTimeout.current );
+		}
+	} );
```

Comment:
We need to add the `[]` for the dependency list, otherwise, this effect will run on every update:

```suggestion
	useEffect(
		() => () => {
			if ( typingTimeout.current ) {
				clearTimeout( typingTimeout.current );
			}
		},
		[]
	);
```

Reply:
Thanks! Sorry I missed that.